src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -275,30 +282,31 @@ private int downloadDataset(List<Map<String, List<DataSet>>> dataSetList) {
   }
 
   private void downloadDataset(Map<String, List<DataSet>> dataSetsPerSample) {
-    for (String sampleCode : dataSetsPerSample.keySet()) {
-      for (DataSet dataset : dataSetsPerSample.get(sampleCode)) {
-        DataSetPermId permID = dataset.getPermId();
+
+    for (Entry<String, List<DataSet>> entry : dataSetsPerSample.entrySet()) {
+      String sampleCode = entry.getKey();
```

Comment:
Looks like this is not needed here. You would also not need to refactor the for-loop before that line

```suggestion
```

Reply:
it is have a look further below 😄